gpkit/tests/t_sub.py

```diff
@@ -138,6 +138,18 @@ def test_signomial(self):
 class TestGPSubs(unittest.TestCase):
     """Test substitution for Model and GP objects"""
 
+    def test_persistence(self):
+        x = gpkit.Variable("x")
+        y = gpkit.Variable("y")
+        ymax = gpkit.Variable("y_{max}", 0.1)
+
+        with gpkit.SignomialsEnabled():
+            m = gpkit.Model(x, [x >= 1-y, y <= ymax])
+            m.substitutions[ymax] = 0.2
+            self.assertAlmostEqual(m.localsolve(verbosity=0)["cost"], 0.8, 3)
+            m = gpkit.Model(x, [x >= 1-y, y <= ymax])
+            self.assertAlmostEqual(m.localsolve(verbosity=0)["cost"], 0.9, 3)
```

Comment:
I'm confused/worried about something.

I tried simplifying the test to a simple GP:
```python
x = Variable("x")
y = Variable("y")
ymin = Variable("y_{min}", 0.1)
m = Model(x, ConstraintSet([x >= y, y >= ymin]))
```
But this does not reproduce the issue (on master). I'm confused why this issue occurs for `SignomialProgram` objects but not plain `Model` objects, since both inherit from `CostedConstraintSet`. Is it possible that, although the change to CostedConstraintSet fixes this issue, there is an additional problem somewhere in SP that we should address?